frontend/workflows/serverexperimentation/src/start-experiment.tsx

```diff
@@ -172,24 +251,75 @@ const StartExperiment: React.FC<StartExperimentProps> = ({
   };
 
   const createExperiment = (data: ExperimentData) => {
+    const isUpstreamEnforcing = data.upstreamClusterType === UpstreamClusterType.INTERNAL;
+    const isTargetingRequests = data.targetType === TargetType.REQUESTS;
+    const isTargetingHosts = data.targetType === TargetType.HOSTS;
+
+    const faultTargeting = {} as IClutch.chaos.serverexperimentation.v1.FaultTargeting;
+    if (isUpstreamEnforcing) {
+      if (isTargetingRequests) {
+        faultTargeting.upstreamEnforcing = {
+          downstreamCluster: {
+            name: data.downstreamCluster,
+          },
+          upstreamCluster: {
+            name: data.upstreamCluster,
+          },
+        };
+      } else {
+        faultTargeting.upstreamEnforcing = {
+          downstreamCluster: {
+            name: data.downstreamCluster,
+          },
+          upstreamPartialSingleCluster: {
+            name: data.upstreamCluster,
+            clusterPercentage: {
+              percentage: isTargetingHosts ? data.hostsPercentage : 100,
+            },
+          },
+        };
+      }
```

Comment:
nit: this could be more concise
```suggestion
      faultTargeting.upstreamEnforcing = {
        downstreamCluster: {
          name: data.downstreamCluster,
        },
      }
      if (isTargetingRequests) {
        faultTargeting.upstreamEnforcing.upstreamCluster = {
          name: data.upstreamCluster,
        };
      } else {
        faultTargeting.upstreamEnforcing.upstreamPartialSingleCluster = {
          name: data.upstreamCluster,
          clusterPercentage: {
            percentage: isTargetingHosts ? data.hostsPercentage : 100,
          },
        };
      }
```